completions/bluetoothctl

```diff
@@ -0,0 +1,46 @@
+# bluetoothctl(1) completion                                    -*- shell-script -*-
+
+_bluetoothctl()
+{
+    local cur=${COMP_WORDS[COMP_CWORD]}
+    local prev=${COMP_WORDS[COMP_CWORD-1]}
+    if [[ "$COMP_CWORD" == "1" ]]; then
+        # Parse help for list of commands and remove ansi colors
+        COMPREPLY=( $( compgen -W "$(bluetoothctl help | sed -n '/  /{s/\x1B\[[0-9;]*[a-zA-Z]//g;s/\x01//g;s/\x02//g;s/ .*//p}')" -- $cur))
```

Comment:
```suggestion
        COMPREPLY=( $( compgen -W "$(bluetoothctl help | sed -n '/  /{s/\x1B\[[0-9;]*[a-zA-Z]//g;s/\x01//g;s/\x02//g;s/ .*//p}')" -- "$cur"))
```
- quoting

```suggestion
        COMPREPLY=( $( compgen -W "$(bluetoothctl help | sed -n '/  /{s/\x1B\[[0-9;]*[a-zA-Z]//g;s/\x01//g;s/\x02//g;s/ .*//p;}')" -- $cur))
```

- [POSIX sed](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/sed.html) requires a semicolon or newline before \<right-brace>.

```suggestion
        COMPREPLY=( $( compgen -W "$(bluetoothctl help | sed -n '/  /{s/'$'\e''\[[0-9;]*[a-zA-Z]//g;s/['$'\x01\x02'']//g;s/ .*//p}')" -- $cur))
```

- AFAIK, POSIX sed doesn't necessarily support the escape sequences like `\x1B`. Here, I'd suggest expanding the sequences using Bash's `$'...'` before passing it to `sed`.

These comments also apply to the later lines that contain the same structure.

Reply:
In that case, it might just be better to not use sed at all. I think this should be doable in bash only.

Reply:
Yeah, I think either works, so you can use what you like.